### PR TITLE
chore: Realign binary naming & ldflags

### PR DESCRIPTION
**Pull request by ocean**
Bring binary naming in Makefile in line with previous releases
(i.e. make a bare binary, named with suffixes, not in a tar.gz file).

Use Makefile foreach to reduce repetition.

Make `ldflags` more similar to Homebrew Go standard `ldflags` so
binaries downloaded from GH or built with Homebrew are more
closely aligned.

### CODE REVIEW

**Comment by AlexSkrypnyk** [comment]
@ocean 
This looks great!

However, I cannot find neither tests for producing these binaries nor direct calls to this command in CI + assertions that they were actually created.

Is it possible for you to add a behat test to run `make cross` and then asserting that a file with a specific extension was created. It won't be asserting that the file is "runnable", but at least we can be sure that the makefile is setup and working correctly.

thanks

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,4 @@ docs/_build
 ahoy
 .idea
 builds/*
+.vscode
```
[the remark this answers is not in the document]
Comment (ocean) [reply .gitignore]: Fair enough, removed.

**Comment by ocean** [comment]
Ok @AlexSkrypnyk - I wrote a BATS test which cross compiles the binaries and then checks to see they exist, good idea 👍

File: tests/cross-compile.bats [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bats
+
+load './test_helpers/bats-support/load'
+load './test_helpers/bats-assert/load'
+
+@test "clean out any cross compiled binaries" {
```
Comment (AlexSkrypnyk) [inline tests/cross-compile.bats:6]: @ocean 
It looks to me that these 3 tests should be in a single test:
1. We clean the binaries and assert that they do not exist
2. We compile them
3. We assert that they were compiled
4. We clean them again and assert that they do not exist

This approach would test the full end-to-end command lifecycle.

Currently, "check cross compiled binaries exist" (3rd test) relies on the result of work of the "cross compile binaries with make" (2nd test), which means that test assets are "bleeding" through tests, making tests not self-contained.

Comment (ocean) [reply tests/cross-compile.bats:6]: Heh, I did have them in two but broke them up more so it was easier to see the tests running.

However, I agree it's better to have it one test and therefore self contained. I looked to see if BATS could output progressive updates to show parts of the test running, but it buffers all output to the end, so no.

**Review by AlexSkrypnyk — APPROVED** [review]
@ocean 
Please merge at your earliest convenience.

**Comment by ocean** [comment]
You're allowed to merge too if you want @AlexSkrypnyk 😜

**Pull request merged**